_posts/2019-10-17-how-to-achieve-cis-benchmark-compliance.adoc

```diff
@@ -0,0 +1,549 @@
+link:https://gruntwork.io/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library/[How to use
+the Gruntwork Infrastructure as Code Library] guide.
+
+[[deployment_walkthrough]]
+== Deployment walkthrough
+This section is a step-by-step walkthrough for achieving compliance with the AWS Foundations Benchmark
+out-of-the-box using the Infrastructure as Code modules from Gruntwork.
+
+[[pre_requisites]]
+=== Pre-requisites
+
+This walkthrough has the following pre-requistes:
+
+Gruntwork Infrastructure as Code Library::
+  This guide uses code from the https://gruntwork.io/infrastructure-as-code-library/[Gruntwork Infrastructure as Code Library], as it
+  implements most of the production-grade design for you out of the box. Make sure to read
+  link:/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library[How to use the Gruntwork Infrastructure as Code Library].
+
+Gruntwork Compliance for CIS AWS Foundations Benchmark::
+  This guide also uses code from the https://gruntwork.io/compliance/cis-aws-foundations-benchmark/[Gruntwork CIS AWS
+  Foundations Benchmark repository], which contains the necessary configurations to achieve compliance.
++
+IMPORTANT: You must be a [js-subscribe-cta]#Gruntwork Compliance subscriber# to access the Gruntwork
+Infrastructure as Code Library and the CIS AWS Foundations Benchmark modules.
+
+Terraform::
+  This guide uses https://www.terraform.io/[Terraform] to define and manage all the infrastructure as code. If
+  you're not familiar with Terraform, check out
+  https://blog.gruntwork.io/a-comprehensive-guide-to-terraform-b3d32832baca[A Comprehensive Guide to Terraform],
+  https://training.gruntwork.io/p/terraform[A Crash Course on Terraform], and
+  link:/guides/foundations/how-to-use-gruntwork-infrastructure-as-code-library[How to Use the Gruntwork
+  Infrastructure as Code Library].
+
+[[manual_steps]]
+=== Manual steps
+A few of the Benchmark recommendations are not achievable via API and require a one-time manual configuration.
+Perform the steps in this section manually.
+
+[[root_mfa]]
+==== Enable MFA for the root account
+Securing the "root" user, or the first user that is created when you set up an AWS account, is one of the
+first actions you should take in any new account. Unfortunately, there is no API or automation available for
+configuring an MFA device for the root user. Follow the manual steps outlined in the
+link:https://docs.aws.amazon.com/IAM/latest/UserGuide/id_root-user.html#id_root-user_manage_mfa[AWS docs].
+
+For the paranoid: configure a hardware MFA device. We suggest using a link:https://www.yubico.com/[Yubikey]
+due to its reputation for strong security characteristics and multitude of form factors. Refer to
+link:https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_mfa_enable_physical.html#enable-hw-mfa-for-root[
+the documentation for more information on using a hardware device with the root user].
```

Comment:
It might be a good idea to include screen shots here to the specific page where you can add MFA in the AWS Console.

Reply:
In general I'm opposed to screenshots as they tend to become out of date and provide little overall value. The AWS docs should be sufficient to guide a user here.